src/tlo/methods/healthsystem.py

```diff
@@ -2370,127 +2369,132 @@ def process_events_mode_2(self, hold_over: List[HSIEventQueueItem]) -> None:
+                                if officer not in set_capabilities_still_available:
+                                    out_of_resources = True
+                            # If officers still available, run event. Note: in current logic, a little
+                            # overtime is allowed to run last event of the day. This seems more realistic
+                            # than medical staff leaving earlier than
+                            # planned if seeing another patient would take them into overtime.
+
+                            if out_of_resources:
+
+                                # Do not run,
+                                # Call did_not_run for the hsi_event
+                                rtn_from_did_not_run = event.did_not_run()
+
+                                # If received no response from the call to did_not_run, or a True signal, then
+                                # add to the hold-over queue.
+                                # Otherwise (disease module returns "FALSE") the event is not rescheduled and
+                                # will not run.
+
+                                if not (rtn_from_did_not_run is False):
+                                    # reschedule event
+                                    # Add the event to the queue:
+                                    hp.heappush(hold_over, next_event_tuple)
+
+                                # Log that the event did not run
+                                self.module.record_hsi_event(
+                                    hsi_event=event,
+                                    actual_appt_footprint=event.EXPECTED_APPT_FOOTPRINT,
+                                    squeeze_factor=squeeze_factor,
+                                    did_run=False,
+                                    priority=_priority
+                                )
 
-                        # Have enough capabilities left to run event
-                        else:
-                            # Notes-to-self: Shouldn't this be done after checking the footprint?
-                            # Compute the bed days that are allocated to this HSI and provide this
-                            # information to the HSI
-                            if sum(event.BEDDAYS_FOOTPRINT.values()):
-                                event._received_info_about_bed_days = \
-                                    self.module.bed_days.issue_bed_days_according_to_availability(
-                                        facility_id=self.module.bed_days.get_facility_id_for_beds(
-                                                                           persons_id=event.target),
-                                        footprint=event.BEDDAYS_FOOTPRINT
-                                    )
+                            # Have enough capabilities left to run event
+                            else:
+                                # Notes-to-self: Shouldn't this be done after checking the footprint?
```

Comment:
The answer to this note is that the checking of bed-days should happen *just before* the HSI is runs. It doesn't gate the HSI running/not running, but should give information that is as up-to-date as possible.